### ISSUE TITLE
Upcoming major releases of Chocolatey products that may impact functionality on the extension

### ISSUE THREAD

**Issue by pauby** [Inedo/inedox-chocolatey#22]
I'm the Technical Engineering Manager at Chocolatey Software, Inc. and I wanted to make sure you are aware that there will be a new major release of Chocolatey products soon. As it is a major release, we are making changes that may have an impact on the Chocolatey extension.

We have beta versions of [Chocolatey CLI](https://community.chocolatey.org/packages/chocolatey) and [Chocolatey GUI](https://community.chocolatey.org/packages/chocolateygui) publicly available that you can test against. 

The release notes for the various products are below (some may or may not be applicable to the module, but I wanted to list them all for completeness):

* [Chocolatey CLI](https://docs.chocolatey.org/en-us/choco/release-notes)
* [Chocolatey Licensed Extension](https://docs.chocolatey.org/en-us/licensed-extension/release-notes)
* [Chocolatey Agent](https://docs.chocolatey.org/en-us/agent/release-notes)
* [Chocolatey GUI](https://docs.chocolatey.org/en-us/chocolatey-gui/release-notes)
* [Chocolatey GUI Licensed Extension](https://docs.chocolatey.org/en-us/chocolatey-gui-licensed-extension/release-notes)

We have no fixed date for release as yet, but it is coming soon.

Apologies for the non-standard issue (I could not find any other way to contact the team). Please let me know if I can help in any way.

**Comment by apxltd** [comment]
Hi @pauby, 

Thanks for the heads up; as an FYI you can contact us via email here: https://inedo.com/company/contact

Alternatively, you can post issues to our public forums: https://forums.inedo.com/

Otherwise it doesn't look like there's a lot changed, but it's hard to follow in the format you shared. That's more of a "Release notes" with very detailed, highly technical changes.

Since this is a major version for you, it'd be very helpful for us (and most certainly all of your users) if there was an aggregate "breaking changes in Chocolatey 2.0" document that listed these more clearly, along with how that might impact users. Here's what we do, for example: https://docs.inedo.com/docs/proget-upgrade-2023

We write these documents months before major versions are released, and we update them as we go along.

Hope that helps,

Alex

**Comment by pauby** [comment]
Hey @apxltd, 

This issue was intended as a courtesy heads-up before the major release, allowing you to investigate the changes, assess the impact and implement any changes needed, before the Chocolatey releases. It's not meant as any kind of 'demand'. If it came across that way, I apologise.

> That's more of a "Release notes" with very detailed, highly technical changes.

I made an assumption that you would be comfortable (and likely prefer) the release notes format / highly technical changes. This is the first major release we have done with breaking changes and the first time we have reached out to any 3rd parties.

>  it'd be very helpful for us (and most certainly all of your users) if there was an aggregate "breaking changes in Chocolatey 2.0" document

That is coming, but won't be published until the release. If you'd prefer to wait for it, keep an eye on our blog, or I can come back and post here. Let me know what you would prefer.

> We write these documents months before major versions are released, and we update them as we go along.

We follow a different process, but sounds like we end up at the same destination.

**Comment by apxltd** [comment]
Thanks @pauby , sorry if my message came across the wrong way! Re-readng it, it's clear my tone was off 🙄

Anyway - good to know that you'll have a "breaking changes" document. We really just use the `choco.exe` in this extension so hopefully nothing major breaks, but we can quickly fix if so.

**Comment by pauby** [comment]
The Chocolatey products were released today. Please refer to the above links for the release notes. Here is some more information that may be helpful:

* [Blog Post](https://blog.chocolatey.org/2023/05/announcing-chocolatey-products-2-and-6/).
* [Guide to upgrading](https://docs.chocolatey.org/en-us/guides/upgrading-to-chocolatey-v2-v6).
* [Highlight of changes](https://docs.chocolatey.org/en-us/choco/new-in-v2).

Please let me know if you need any more details.

**Comment by apxltd** [comment]
@pauby congratulations!

Great to see NuGet v3 support; we'll enable that option for our Choco feeds, mentioning it's only supported on client 2.0+
